solr/core/src/java/org/apache/solr/util/circuitbreaker/CircuitBreakerRegistry.java

```diff
@@ -97,4 +112,34 @@ public static String toErrorMessage(List<CircuitBreaker> circuitBreakerList) {
   public boolean isEnabled(SolrRequestType requestType) {
     return circuitBreakerMap.containsKey(requestType);
   }
+
+  @Override
+  public void close() throws IOException {
+    synchronized (circuitBreakerMap) {
+      final AtomicBoolean closeFailed = new AtomicBoolean(false);
+      circuitBreakerMap
+          .values()
+          .forEach(
+              list ->
+                  list.forEach(
+                      it -> {
+                        try {
+                          if (log.isDebugEnabled()) {
+                            log.debug(
+                                "Closed circuit breaker {} for request type(s) {}",
+                                it.getClass().getSimpleName(),
+                                it.getRequestTypes());
+                          }
+                          it.close();
+                        } catch (IOException e) {
+                          log.error("Failed to close circuit breaker", e);
+                          closeFailed.set(true);
+                        }
+                      }));
+      circuitBreakerMap.clear();
+      if (closeFailed.get()) {
+        throw new IOException("Failed to close one or more circuit breakers");
```

Comment:
minor: swapping out of the atomic boolean for integer could allow specific wording.
```suggestion
        throw new IOException("Failed to close " + closeFailedCounter.get() + " circuit breakers");
```